source/common/router/router.cc

```diff
@@ -804,6 +832,15 @@ void Filter::onPerTryTimeout(UpstreamRequest& upstream_request) {
     upstream_request.upstream_host_->stats().rq_timeout_.inc();
   }
 
+  if (cluster_->timeoutBudgetStats().has_value()) {
+    Event::Dispatcher& dispatcher = callbacks_->dispatcher();
+    std::chrono::milliseconds response_time = std::chrono::duration_cast<std::chrono::milliseconds>(
+        dispatcher.timeSource().monotonicTime() - downstream_request_complete_time_);
+
+    cluster_->timeoutBudgetStats()->upstream_rq_timeout_budget_percent_used_.recordValue(
+        percentageOfTimeout(response_time, timeout_.global_timeout_));
+  }
```

Comment:
I'm confused. This is tracking the global timeout in the per-try timeout case?

Reply:
This was me messing up and putting this in the wrong place.  Will fix.